src/core/qgspallabeling.cpp

```diff
@@ -3944,7 +3944,7 @@ QgsGeometry* QgsPalLabeling::prepareGeometry( const QgsGeometry* geometry, QgsRe
     clonedGeometry.reset( geom );
   }
 
-  if ( clipGeometry && !clipGeometry->contains( geom ) )
+  if ( clipGeometry && !clipGeometry->boundingBox().contains( geom->boundingBox() ) )
```

Comment:
@nyalldawson sorry, I mixed the code in my mind, my apologies, you are right!

Thanks for your pacience :-)